lib/lightning_web/live/job_live/job_builder.ex

```diff
@@ -45,6 +46,18 @@ defmodule LightningWeb.JobLive.JobBuilder do
     )
   end
 
+  defp is_deletable(job_id) do
+    Jobs.get_job!(job_id) |> Jobs.get_downstream_jobs_for() |> Enum.count() == 0
+  end
+
+  defp delete_job_title(job_id) do
+    if is_deletable(job_id) do
+      "Delete this job"
+    else
+      "Impossible to delete upstream jobs. Please delete all associated downstream jobs first."
+    end
+  end
+
```

Comment:
I have a concern here about performance, see: https://hexdocs.pm/phoenix_live_view/assigns-eex.html#change-tracking

I think that a `@is_deletable` assign should be added to `update/3`, without that we run the risk of the database being called with every render (i.e. `handle_event/3`).